src/refmvs.rs

```diff
@@ -1136,31 +1136,27 @@ pub(crate) unsafe fn rav1d_refmvs_find(
 // cache the current tile/sbrow (or frame/sbrow)'s projectable motion vectors
 // into buffers for use in future frame's temporal MV prediction
 pub(crate) unsafe fn rav1d_refmvs_save_tmvs(
-    dsp: *const Rav1dRefmvsDSPContext,
-    rt: *const refmvs_tile,
+    dsp: &Rav1dRefmvsDSPContext,
+    rt: &refmvs_tile,
     col_start8: c_int,
-    mut col_end8: c_int,
+    col_end8: c_int,
     row_start8: c_int,
-    mut row_end8: c_int,
+    row_end8: c_int,
 ) {
-    let rf: *const refmvs_frame = (*rt).rf;
-    if !(row_start8 >= 0) {
-        unreachable!();
-    }
-    if !((row_end8 - row_start8) as c_uint <= 16 as c_uint) {
-        unreachable!();
-    }
-    row_end8 = cmp::min(row_end8, (*rf).ih8);
-    col_end8 = cmp::min(col_end8, (*rf).iw8);
-    let stride: ptrdiff_t = (*rf).rp_stride;
-    let ref_sign: *const u8 = ((*rf).mfmv_sign).as_ptr();
-    let rp: *mut refmvs_temporal_block = (*rf).rp.offset(row_start8 as isize * stride);
-
-    (*dsp).save_tmvs.expect("non-null function pointer")(
+    let rf = &*rt.rf;
+    assert!(row_start8 >= 0);
+    assert!((row_end8 - row_start8) as c_uint <= 16);
```

Comment:
I think the cast here is redundant:

```suggestion
    assert!((row_end8 - row_start8) <= 16);
```

Reply:
C deliberately does the unsigned cast here.  I think they're folding in an `assert!(_ >= 0)` here.